djangocms_googlemap/models.py

```diff
@@ -1,90 +1,287 @@
+# Add additional choices through the ``settings.py``.
+def get_templates():
+    choices = [
+        ('default', _('Default')),
+    ]
+    choices += getattr(
+        settings,
+        'DJANGOCMS_GOOGLEMAP_TEMPLATES',
+        [],
+    )
+    return choices
 
-    style = models.TextField(_("custom map style"), blank=True,
-        help_text=_('Provide a valid JSON configuration (escaped). See '
-            'developers.google.com/maps/documentation/javascript/styling'))
 
-    ZOOM_LEVELS = map(lambda c: (c, str(c)), range(22))
+@python_2_unicode_compatible
+class GoogleMap(CMSPlugin):
+    """
+    Renders the Google Maps wrapper
+    """
+    template = models.CharField(
+        verbose_name=_('Template'),
+        choices=get_templates(),
+        default=get_templates()[0][0],
+        max_length=255,
+    )
+    title = models.CharField(
+        verbose_name=_('Map title'),
+        max_length=255,
+        blank=True,
+    )
+    width = models.CharField(
+        verbose_name=_('Width'),
+        max_length=6,
+        default='100%',
+        help_text=_('Width of the map (e.g. "100%" or "400px").'),
+    )
+    height = models.CharField(
+        verbose_name=_('Height'),
+        max_length=6,
+        default='400px',
+        help_text=_('Height of the map (e.g. "400px").'),
+    )
+    style = models.TextField(
+        verbose_name=_('Map styling'),
+        blank=True,
+        help_text=_('Provide a valid JSON configuration (escaped). See '
+            'http://developers.google.com/maps/documentation/javascript/styling'),
+    )
+    lat = models.FloatField(
+        verbose_name=_('Latitude (lat)'),
+        null=True,
+        blank=True,
+        help_text=_('Default GPS latitute (e.g. "46.947973").'),
+    )
+    lng = models.FloatField(
+        verbose_name=_('Longitude (lng)'),
+        null=True,
+        blank=True,
+        help_text=_('Default GPS longitude (e.g. "7.447446").'),
+    )
     zoom = models.PositiveSmallIntegerField(
-        _("zoom level"), choices=ZOOM_LEVELS, default=13)
-
-    lat = models.DecimalField(
-        _('latitude'), max_digits=10, decimal_places=6, null=True, blank=True,
-        help_text=_('Use latitude & longitude to fine tune the map position.'))
+        verbose_name=_('Zoom level'),
+        choices=ZOOM_LEVEL_CHOICES,
+        default=13,
+    )
 
-    lng = models.DecimalField(
-        _('longitude'), max_digits=10, decimal_places=6, null=True, blank=True)
-
-    route_planer_title = models.CharField(
-        _("route planner title"), max_length=150, blank=True, null=True,
-        default=_('Calculate your fastest way to here'))
-
-    route_planer = models.BooleanField(_("route planner"), default=False)
+    # control settings
+    map_type_control = models.CharField(
+        verbose_name=_('Map type control'),
+        choices=MAP_TYPE_CHOICES,
+        default=MAP_TYPE_CHOICES[0][0],
+        max_length=255,
+    )
```

Comment:
The user is choosing the type of map, not the type of map control - so it should be "Map type".